modules/filestore/amazonS3/delete.go

```diff
@@ -0,0 +1,41 @@
+package amazonS3
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+func (a *AmazonS3) DeleteFile(ctx context.Context, project, path string) error {
+	svc := s3.New(a.session)
+	_, err := svc.DeleteObject(&s3.DeleteObjectInput{Bucket: aws.String(project), Key: aws.String(path)})
+	if err != nil {
+		return err
+	}
+
+	err = svc.WaitUntilObjectNotExists(&s3.HeadObjectInput{
+		Bucket: aws.String(project),
+		Key:    aws.String(project + path),
+	})
+	return err
+}
+
+func (a *AmazonS3) DeleteDir(ctx context.Context, project, path string) {
+	svc := s3.New(a.session)
+
+	// Setup BatchDeleteIterator to iterate through a list of objects.
+	iter := s3manager.NewDeleteListIterator(svc, &s3.ListObjectsInput{
```

Comment:
ok i have tested only for multiple files under directory, will test that too
the limit is 1000 and u have to use some pagination function to get rid of that limit

Reply:
Hmm. add the todo. // TODO: Consider AWS operation limit

we won't implement pagination right or anything right now. That TODO will serve as a reminder that it needs to be done. Cool?

Reply:
i have tested it deletes everyting 
suppose path : sharad/
it will delete everything under sharad including the directory sharad will be deleted is that ok
or u don't want the directory sharad to be deleted ?

Reply:
reply

Reply:
Everything should be deleted including the directory sharad